### PR TITLE
Tor: Move from extra, add 64bit and portable

### PR DESCRIPTION
**Pull request by linsui**
https://github.com/lukesampson/scoop-extras/pull/2471

### CODE REVIEW

**Comment by linsui** [comment]
Why `$dir` in `pre_install` is `tor/0.4.0.5` ? I expect it is `C:\Users\username\scoop\apps\$app\current` as doc says.

File: bucket/tor.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            "hash": "76761d916772f6c5c3c43dbe4d9ca846fe6526fcbe8974b88b11a00a196e609d"
+        },
+        "32bit": {
+            "url": "https://dist.torproject.org/torbrowser/8.5.4/tor-win32-0.4.0.5.zip",
+            "hash": "8d411a394597bab5ec6fe446a2cda21279c830c5fdd19c3de82aa1028ffd7e96"
+        }
+    },
+    "bin": [
+        [
+            "Tor/tor.exe",
```
Comment (Ash258) [inline bucket/tor.json]: Use backslashes

**Comment by Ash258** [comment]
I do not know what variable you were trying to use, but $dir is working as expected.

image

File: bucket/tor.json [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        "torrc"
+    ],
+    "pre_install": [
+        "$content = @(",
+        "    \"DataDirectory `\"$dir\\Data\\Data`\"\"",
+        "    \"GeoIPFile `\"$dir\\Data\\Tor\\geoip`\"\"",
+        "    \"GeoIPv6File `\"$dir\\Data\\Tor\\geoip6`\"\"",
+        ")",
+        "if (!(Test-Path \"$persist_dir\\torrc\")) {",
+        "    New-Item -Force -Path \"$dir\" -Name torrc | Out-Null",
```
Comment (Ash258) [inline bucket/tor.json]: You do not need to create file. Just add `-Force` to Add-Content

**Comment by linsui** [comment]
I expect it is the `current` dir but I get `0.4.0.5` in `torrc` .

**Comment by Ash258** [comment]
That is correct.

**Comment by linsui** [comment]
Then how can I get the `current` path? If the `0.4.0.5` is write into `torrc` and be persisted, it can't work after update.

**Comment by Ash258** [comment]
In `pre_install` there is no current folder. Current folder is created after installer

**Comment by Ash258** [comment]
`$current = current_dir $dir`

**Comment by linsui** [comment]
I got. Thanks!

**Pull request merged**